# Dataframe Manipulation Warmup

In [1]:
import numpy as np
import pandas as pd

np.random.seed(406)

n = 5000
df = pd.DataFrame({
    'favorite_animal': np.random.choice(['cat', 'dog', 'frog', 'lemur', 'panda'], n),
    'favorite_vegetable': np.random.choice(['brussel sprouts', 'potato', 'squash'], n),
    'favorite_fruit': np.random.choice(['banana', 'apple', 'blueberries'], n),
    'wears_glasses': np.random.choice(['yes', 'no'], n),
    'netflix_consumption': np.random.normal(10, 2, n),
    'open_browser_tabs': np.random.randint(2, 90, n),
})


In [3]:
df.head()

favorite_animal favorite_vegetable favorite_fruit wears_glasses  \
0           lemur             potato          apple           yes   
1           panda             potato          apple           yes   
2             cat             squash    blueberries           yes   
3           lemur             squash          apple            no   
4             dog    brussel sprouts          apple           yes   

   netflix_consumption  open_browser_tabs  
0             8.313351                 44  
1            11.801073                 10  
2            10.105141                 35  
3            11.024605                 70  
4             6.732698                 73

In [9]:

print(f'1. {df.netflix_consumption.max()}')
print(f'2. {(df.wears_glasses == "yes").sum()}, {((df.wears_glasses == "yes").sum()/len(df)):.2%}')
print(f'3. {(df.favorite_animal == "dog").sum()}')
print(f'4. {df.favorite_animal.mode()}')

sprout_eaters = df[df.favorite_vegetable == 'brussel sprouts']
print(f'5. {sprout_eaters.netflix_consumption.mean()}')

four_eyes = df[df.wears_glasses == "yes"]
print(f'6. {four_eyes[four_eyes.open_browser_tabs > 40].favorite_fruit.mode()}')




1. 17.534818515438925
2. 2555, 51.10%
3. 1002
4. 0    lemur
dtype: object
5. 10.00847154798366
6. 0    blueberries
dtype: object


1028

1. What is the highest amount of netflix consumption? `17.535`
2. How many people wear glasses? What percentage of people is this? `2555`, `.511`
3. How many people's favorite animal is a dog? `1002`
4. What is the most common favorite animal? `lemur`
5. What is the average netflix consumption for people that prefer brussel sprouts? `10.008`
6. What is the most common favorite fruit for people who wear glasses and have more than 40 open browser tabs? `blueberries`

In [7]:
sprout_eaters.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1696 entries, 4 to 4992
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   favorite_animal      1696 non-null   object 
 1   favorite_vegetable   1696 non-null   object 
 2   favorite_fruit       1696 non-null   object 
 3   wears_glasses        1696 non-null   object 
 4   netflix_consumption  1696 non-null   float64
 5   open_browser_tabs    1696 non-null   int64  
dtypes: float64(1), int64(1), object(4)
memory usage: 92.8+ KB


In [3]:
df[(df.wears_glasses == 'yes') & (df.open_browser_tabs > 40)].favorite_fruit.value_counts().sort_values(ascending=False)

blueberries    498
apple          477
banana         472
Name: favorite_fruit, dtype: int64